### PR TITLE
Fix type error, where whitelist gets checked against none in mesos_tools

### PR DESCRIPTION
**Pull request by matthewbentley**
Also fix type error when logging errors

### CODE REVIEW

File: paasta_tools/smartstack_tools.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def get_smartstack_replication_for_attribute(attribute, service, namespace, blac
     replication_info = {}
     filtered_slaves = mesos_tools.get_all_slaves_for_blacklist_whitelist(
         blacklist=blacklist,
-        whitelist=[],
+        whitelist=None,
```
Comment (Rob-Johnson) [inline paasta_tools/smartstack_tools.py]: isn't it the code on the other side thats wrong? i.e. why aren't we passing [] here and the ``get_all_slaves_for_blacklist_whitelist`` function dealing with it correctly?

Comment (matthewbentley) [reply paasta_tools/smartstack_tools.py]: `DeployWhitelist` is an `Optional`, so `None` is fine: https://github.com/Yelp/paasta/blob/master/paasta_tools/utils.py#L151

Comment (matthewbentley) [reply paasta_tools/smartstack_tools.py]: actually, I think the type in utils.py is wrong

Comment (matthewbentley) [reply paasta_tools/smartstack_tools.py]: ...or the type is right, but the docstring on `slave_passes_whitelist` is wrong

**Pull request merged**